doc/classes/VisualShaderConversionPlugin.xml

```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<class name="VisualShaderConversionPlugin" inherits="EditorResourceConversionPlugin" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../class.xsd">
+	<brief_description>
+		Plugin for converting from [VisualShader] to [Shader].
+	</brief_description>
+	<description>
+		[VisualShaderConversionPlugin] is invoked when the context menu is brought up for a [VisualShader] in the editor inspector.
+	</description>
+	<tutorials>
+	</tutorials>
+	<methods>
+		<method name="convert" qualifiers="const">
+			<return type="Resource" />
+			<param index="0" name="resource" type="Resource" />
+			<description>
+				Takes an input [Resource] expected to be of type [VisualShader] and converts it to [Shader] in [method converts_to]. The returned [Resource] is the resulting [Shader] of the conversion, and the input [Resource] remains unchanged.
+			</description>
+		</method>
+		<method name="converts_to" qualifiers="const">
+			<return type="String" />
+			<description>
+				Returns "Shader" which is the class name [Shader] this plugin converts the source resource [VisualShader] to.
+			</description>
+		</method>
+		<method name="handles" qualifiers="const">
+			<return type="bool" />
+			<param index="0" name="resource" type="Resource" />
+			<description>
+				Called to determine whether a particular [Resource] can be converted to the [Shader] resource type by this plugin. Will return false if the particular [Resource] is not of type [VisualShader].
+			</description>
+		</method>
+	</methods>
```

Comment:
What's the point of exposing these methods if the result is known? Especially `converts_to()` - what is it useful for, in custom plugin context?

Reply:
We don't have to.  I was just taking a holistic approach and exposing the public methods so gdscript, csharp, and gdextension can interact with the VisualShaderConversionPlugin.  I found the ``convert`` method useful and figured I or others may find use for the other methods in the future.  But if you want, I can remove those methods.  I personally only need the ``convert`` method at this very moment.

Reply:
We shouldn't expose more methods than needed. If someone finds a use-case in the future, they can be exposed, but unexposing something is not possible because of compatibility.

Reply:
Ah I see.  I will go remove these methods then :)

Reply:
Actually, this brings front a new question:

Do we want this PR to go through?  Do we want to keep these methods internal to the engine?  And instead implement this feature proposal here? https://github.com/godotengine/godot-proposals/issues/11561#issuecomment-2659431002

Reply:
Well, if the proposal covers the same use-case, there is no need to expose it.

Reply:
It would for me yes.  If this is the case, I'm in favor of just leaving this change in my fork for now while an official solution for that proposal is worked on.  This way we don't unnecessarily expose methods and maintain the ability to make breaking changes to them in the future.

If the proposal never receives a solution and / or if this change becomes to challenging to maintain in parallel in our fork (which I highly doubt it will, its a simple change), I'm in favor of just closing this PR.